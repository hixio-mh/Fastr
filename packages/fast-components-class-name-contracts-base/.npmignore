# Source files
src/

# Configs
tsconfig.json
tslint.json