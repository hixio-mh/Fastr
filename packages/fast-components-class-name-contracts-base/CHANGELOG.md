# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [3.1.0](https://github.com/Microsoft/fast-dna/compare/@microsoft/fast-components-class-name-contracts-base@3.0.1...@microsoft/fast-components-class-name-contracts-base@3.1.0) (2018-10-27)


### Features

* add radio component ([#884](https://github.com/Microsoft/fast-dna/issues/884)) ([f3cf735](https://github.com/Microsoft/fast-dna/commit/f3cf735))





## [3.0.1](https://github.com/Microsoft/fast-dna/compare/@microsoft/fast-components-class-name-contracts-base@3.0.0...@microsoft/fast-components-class-name-contracts-base@3.0.1) (2018-10-09)

**Note:** Version bump only for package @microsoft/fast-components-class-name-contracts-base





# [3.0.0](https://github.com/Microsoft/fast-dna/compare/@microsoft/fast-components-class-name-contracts-base@2.2.0...@microsoft/fast-components-class-name-contracts-base@3.0.0) (2018-10-06)


### Bug Fixes

* allow nested at rules in styles without needing to define it in the class name contract ([#985](https://github.com/Microsoft/fast-dna/issues/985)) ([6479d4c](https://github.com/Microsoft/fast-dna/commit/6479d4c))


### chore

* remove the 'I' from interfaces ([#997](https://github.com/Microsoft/fast-dna/issues/997)) ([d924df8](https://github.com/Microsoft/fast-dna/commit/d924df8))


### Features

* add card as a new component ([#984](https://github.com/Microsoft/fast-dna/issues/984)) ([9514199](https://github.com/Microsoft/fast-dna/commit/9514199))
* **class names:** update all class names across base and MSFT to align with naming standards ([#893](https://github.com/Microsoft/fast-dna/issues/893)) ([de974ab](https://github.com/Microsoft/fast-dna/commit/de974ab))
* update class-name contracts to be entirely optional ([#959](https://github.com/Microsoft/fast-dna/issues/959)) ([58e0421](https://github.com/Microsoft/fast-dna/commit/58e0421))


### BREAKING CHANGES

* Interfaces have been renamed to remove the "I".





<a name="2.2.0"></a>
# 2.2.0 (2018-09-11)


### Features

* add contrast based color system ([#810](https://github.com/Microsoft/fast-dna/issues/810)) ([5ec457c](https://github.com/Microsoft/fast-dna/commit/5ec457c))
* **progress:** add progress component ([#835](https://github.com/Microsoft/fast-dna/issues/835)) ([44d9bb1](https://github.com/Microsoft/fast-dna/commit/44d9bb1))



<a name="2.1.0"></a>
# 2.1.0 (2018-08-29)


### Features

* update Lerna to ^3.0.0 ([#795](https://github.com/Microsoft/fast-dna/issues/795)) ([9ce9a56](https://github.com/Microsoft/fast-dna/commit/9ce9a56))
* upgrade to TypeScript 3.0.0 ([#793](https://github.com/Microsoft/fast-dna/issues/793)) ([e203e86](https://github.com/Microsoft/fast-dna/commit/e203e86))
* **fast-components-react-base:** add callback to horizontal overflow to return and object that informs scroll start and end ([#797](https://github.com/Microsoft/fast-dna/issues/797)) ([37975f3](https://github.com/Microsoft/fast-dna/commit/37975f3))
* **fast-components-react-base:** add tabs component ([#761](https://github.com/Microsoft/fast-dna/issues/761)) ([24a5bb3](https://github.com/Microsoft/fast-dna/commit/24a5bb3))



<a name="2.0.0-corrected"></a>
# 2.0.0-corrected (2018-08-03)


### Bug Fixes

* travis-CI build-break ([#336](https://github.com/Microsoft/fast-dna/issues/336)) ([bffbf5e](https://github.com/Microsoft/fast-dna/commit/bffbf5e))
* **fast-components-react-msft:** fixes error running jest with components that require chroma ([#687](https://github.com/Microsoft/fast-dna/issues/687)) ([140457c](https://github.com/Microsoft/fast-dna/commit/140457c))


### Features

* add hypertext component ([#210](https://github.com/Microsoft/fast-dna/issues/210)) ([9e363ff](https://github.com/Microsoft/fast-dna/commit/9e363ff))
* **checkbox:** update styling and incorrect states ([#371](https://github.com/Microsoft/fast-dna/issues/371)) ([45cbe3c](https://github.com/Microsoft/fast-dna/commit/45cbe3c))
* Forked Class name contracts so we can have one for Base and one for MSFT ([#262](https://github.com/Microsoft/fast-dna/issues/262)) ([a4c54c0](https://github.com/Microsoft/fast-dna/commit/a4c54c0))
* **checkbox:** add new component with styles ([#252](https://github.com/Microsoft/fast-dna/issues/252)) ([3ad3988](https://github.com/Microsoft/fast-dna/commit/3ad3988))
* update code coverage on travis ([#330](https://github.com/Microsoft/fast-dna/issues/330)) ([63ab4f4](https://github.com/Microsoft/fast-dna/commit/63ab4f4))
* **detail view:** add detail view ([#470](https://github.com/Microsoft/fast-dna/issues/470)) ([665b871](https://github.com/Microsoft/fast-dna/commit/665b871))
* **dialog:** add dialog as a new component ([#752](https://github.com/Microsoft/fast-dna/issues/752)) ([2864021](https://github.com/Microsoft/fast-dna/commit/2864021))
* **fast-components-react-base:** add horizontal overflow ([#739](https://github.com/Microsoft/fast-dna/issues/739)) ([c6b0ebf](https://github.com/Microsoft/fast-dna/commit/c6b0ebf))
* **fast-css-editor-react:** add default editor component and position component ([#636](https://github.com/Microsoft/fast-dna/issues/636)) ([72037a8](https://github.com/Microsoft/fast-dna/commit/72037a8))
* **form generator:** updates styles found in configuration pane ([#420](https://github.com/Microsoft/fast-dna/issues/420)) ([919121b](https://github.com/Microsoft/fast-dna/commit/919121b))
* **label:** add new component and msft styles ([#265](https://github.com/Microsoft/fast-dna/issues/265)) ([0328028](https://github.com/Microsoft/fast-dna/commit/0328028))





<a name="2.1.0"></a>
# [2.1.0](https://github.com/Microsoft/fast-dna/compare/v2.0.0-corrected...v2.1.0) (2018-08-29)


### Features

* update Lerna to ^3.0.0 ([#795](https://github.com/Microsoft/fast-dna/issues/795)) ([9ce9a56](https://github.com/Microsoft/fast-dna/commit/9ce9a56))
* upgrade to TypeScript 3.0.0 ([#793](https://github.com/Microsoft/fast-dna/issues/793)) ([e203e86](https://github.com/Microsoft/fast-dna/commit/e203e86))
* **fast-components-react-base:** add callback to horizontal overflow to return and object that informs scroll start and end ([#797](https://github.com/Microsoft/fast-dna/issues/797)) ([37975f3](https://github.com/Microsoft/fast-dna/commit/37975f3))
* **fast-components-react-base:** add tabs component ([#761](https://github.com/Microsoft/fast-dna/issues/761)) ([24a5bb3](https://github.com/Microsoft/fast-dna/commit/24a5bb3))





<a name="2.0.0"></a>
# [2.0.0](https://github.com/Microsoft/fast-dna/compare/v1.6.0...v2.0.0) (2018-08-02)


### Bug Fixes

* **fast-components-react-msft:** fixes error running jest with components that require chroma ([#687](https://github.com/Microsoft/fast-dna/issues/687)) ([140457c](https://github.com/Microsoft/fast-dna/commit/140457c))


### Features

* **detail view:** add detail view ([#470](https://github.com/Microsoft/fast-dna/issues/470)) ([665b871](https://github.com/Microsoft/fast-dna/commit/665b871))
* **dialog:** add dialog as a new component ([#752](https://github.com/Microsoft/fast-dna/issues/752)) ([2864021](https://github.com/Microsoft/fast-dna/commit/2864021))
* **fast-components-react-base:** add horizontal overflow ([#739](https://github.com/Microsoft/fast-dna/issues/739)) ([c6b0ebf](https://github.com/Microsoft/fast-dna/commit/c6b0ebf))
* **fast-css-editor-react:** add default editor component and position component ([#636](https://github.com/Microsoft/fast-dna/issues/636)) ([72037a8](https://github.com/Microsoft/fast-dna/commit/72037a8))




<a name="1.9.0"></a>
# [1.9.0](https://github.com/Microsoft/fast-dna/compare/v1.6.0...v1.9.0) (2018-07-14)


### Bug Fixes

* **fast-components-react-msft:** fixes error running jest with components that require chroma ([#687](https://github.com/Microsoft/fast-dna/issues/687)) ([140457c](https://github.com/Microsoft/fast-dna/commit/140457c))


### Features

* **detail view:** add detail view ([#470](https://github.com/Microsoft/fast-dna/issues/470)) ([665b871](https://github.com/Microsoft/fast-dna/commit/665b871))
* **fast-css-editor-react:** add default editor component and position component ([#636](https://github.com/Microsoft/fast-dna/issues/636)) ([72037a8](https://github.com/Microsoft/fast-dna/commit/72037a8))




<a name="1.8.0"></a>
# [1.8.0](https://github.com/Microsoft/fast-dna/compare/v1.6.0...v1.8.0) (2018-06-12)




<a name="1.7.0"></a>
# [1.7.0](https://github.com/Microsoft/fast-dna/compare/v1.6.0...v1.7.0) (2018-06-01)


### Features

* **detail view:** add detail view ([#470](https://github.com/Microsoft/fast-dna/issues/470)) ([665b871](https://github.com/Microsoft/fast-dna/commit/665b871))




<a name="1.6.0"></a>
# [1.6.0](https://github.com/Microsoft/fast-dna/compare/v1.2.0...v1.6.0) (2018-05-16)




<a name="1.5.0"></a>
# [1.5.0](https://github.com/Microsoft/fast-dna/compare/v1.2.0...v1.5.0) (2018-05-16)


### Features

* **form generator:** updates styles found in configuration pane ([#420](https://github.com/Microsoft/fast-dna/issues/420)) ([919121b](https://github.com/Microsoft/fast-dna/commit/919121b))




<a name="1.4.0"></a>
# [1.4.0](https://github.com/Microsoft/fast-dna/compare/v1.2.0...v1.4.0) (2018-05-14)


### Features

* **form generator:** updates styles found in configuration pane ([#420](https://github.com/Microsoft/fast-dna/issues/420)) ([919121b](https://github.com/Microsoft/fast-dna/commit/919121b))




<a name="1.2.0"></a>
# 1.2.0 (2018-05-10)


### Features

* **checkbox:** update styling and incorrect states ([#371](https://github.com/Microsoft/fast-dna/issues/371)) ([45cbe3c](https://github.com/Microsoft/fast-dna/commit/45cbe3c))




<a name="1.1.0"></a>
# 1.1.0 (2018-05-09)


### Bug Fixes

* travis-CI build-break ([#336](https://github.com/Microsoft/fast-dna/issues/336)) ([bffbf5e](https://github.com/Microsoft/fast-dna/commit/bffbf5e))


### Features

* add hypertext component ([#210](https://github.com/Microsoft/fast-dna/issues/210)) ([9e363ff](https://github.com/Microsoft/fast-dna/commit/9e363ff))
* Forked Class name contracts so we can have one for Base and one for MSFT ([#262](https://github.com/Microsoft/fast-dna/issues/262)) ([a4c54c0](https://github.com/Microsoft/fast-dna/commit/a4c54c0))
* **checkbox:** add new component with styles ([#252](https://github.com/Microsoft/fast-dna/issues/252)) ([3ad3988](https://github.com/Microsoft/fast-dna/commit/3ad3988))
* **label:** add new component and msft styles ([#265](https://github.com/Microsoft/fast-dna/issues/265)) ([0328028](https://github.com/Microsoft/fast-dna/commit/0328028))
* update code coverage on travis ([#330](https://github.com/Microsoft/fast-dna/issues/330)) ([63ab4f4](https://github.com/Microsoft/fast-dna/commit/63ab4f4))
