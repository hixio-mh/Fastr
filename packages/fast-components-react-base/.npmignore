# Tests
__test__/
*.spec.*
*.test.*

# Source files
app/
coverage/
src/
www/

# babel config
babel.config.js
tsconfig.json
tslint.json
webpack.config.js