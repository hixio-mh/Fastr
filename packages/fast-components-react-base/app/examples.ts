import AutoSuggestExamples from "../src/auto-suggest/examples.data";
export { AutoSuggestExamples };

import BadgeExamples from "../src/badge/examples.data";
export { BadgeExamples };

import BreadcrumbExamples from "../src/breadcrumb/examples.data";
export { BreadcrumbExamples };

import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CardExamples from "../src/card/examples.data";
export { CardExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import ContextMenu from "../src/context-menu/examples.data";
export { ContextMenu };

import ContextMenuItem from "../src/context-menu-item/examples.data";
export { ContextMenuItem };

import DialogExamples from "../src/dialog/examples.data";
export { DialogExamples };

import DividerExamples from "../src/divider/examples.data";
export { DividerExamples };

import HorizontalOverflowExamples from "../src/horizontal-overflow/examples.data";
export { HorizontalOverflowExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import ListboxExamples from "../src/listbox/examples.data";
export { ListboxExamples };

import ListboxItemExamples from "../src/listbox-item/examples.data";
export { ListboxItemExamples };

import NumberFieldExamples from "../src/number-field/examples.data";
export { NumberFieldExamples };

import ProgressExamples from "../src/progress/examples.data";
export { ProgressExamples };

import RadioExamples from "../src/radio/examples.data";
export { RadioExamples };

import SelectExamples from "../src/select/examples.data";
export { SelectExamples };

import SliderExamples from "../src/slider/examples.data";
export { SliderExamples };

import SliderTrackItemExamples from "../src/slider-track-item/examples.data";
export { SliderTrackItemExamples };

import TabsExamples from "../src/tabs/examples.data";
export { TabsExamples };

import TextAreaExamples from "../src/text-area/examples.data";
export { TextAreaExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
