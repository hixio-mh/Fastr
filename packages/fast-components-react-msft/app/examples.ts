import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CallToActionExamples from "../src/call-to-action/examples.data";
export { CallToActionExamples };

import CaptionExamples from "../src/caption/examples.data";
export { CaptionExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import Dialog from "../src/dialog/examples.data";
export { Dialog };

import DividerExample from "../src/divider/examples.data";
export { DividerExample };

import FlipperExample from "../src/flipper/examples.data";
export { FlipperExample };

import HeadingExample from "../src/heading/examples.data";
export { HeadingExample };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import MetatextExamples from "../src/metatext/examples.data";
export { MetatextExamples };

import ParagraphExamples from "../src/paragraph/examples.data";
export { ParagraphExamples };

import ProgressExamples from "../src/progress/examples.data";
export { ProgressExamples };

import SubheadingExamples from "../src/subheading/examples.data";
export { SubheadingExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
