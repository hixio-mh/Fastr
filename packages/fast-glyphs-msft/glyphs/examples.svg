<svg width="16" height="16" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="Canvas">
<g id="Group">
<g id="Vector">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.3 5.8L 2.3 3.9L 0.6 4.8L -3.8147e-07 3.8L 1.7 2.9L 0.1 2L 0.7 1L 2.2 1.9L 2.2 2.38419e-08L 3.3 2.38419e-08L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(10.3999 1.09998)"/>
</g>
<g id="Vector_2">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.2 5.8L 2.2 3.9L 0.6 4.8L 4.76837e-08 3.8L 1.6 2.9L 4.76837e-08 2L 0.6 1L 2.2 1.9L 2.2 -1.90735e-07L 3.3 -1.90735e-07L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(1.19995 10.2)"/>
</g>
<g id="Vector_3">
<path d="M 0 8L 0 0L 8 0L 8 8L 0 8ZM 1 1L 1 7L 7 7L 7 1L 1 1ZM 14.4 13.1L 16 14L 15.4 15L 13.8 14.1L 13.8 16L 12.7 16L 12.7 14.1L 11 15.1L 10.4 14.1L 12 13.2L 10.4 12.3L 11 11.3L 12.6 12.2L 12.6 10.3L 13.7 10.3L 13.7 12.2L 15.3 11.3L 15.9 12.3L 14.4 13.1Z"/>
</g>
</g>
</g>
</svg>