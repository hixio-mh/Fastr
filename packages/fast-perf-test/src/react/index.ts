import React from "react";
import ReactDOM from "react-dom";
import { Checkbox, TextField } from "@microsoft/fast-components-react-msft";

const root = document.createElement("div");
document.body.appendChild(root);

function getQueryVariable(variable) {
    var query = window.location.search.substring(1);
    var vars = query.split("&");
    for (var i = 0; i < vars.length; i++) {
        var pair = vars[i].split("=");
        if (decodeURIComponent(pair[0]) == variable) {
            return decodeURIComponent(pair[1]);
        }
    }
}

const count = getQueryVariable("count");

function renderItems() {
    return React.createElement(
        React.Fragment,
        {},
        new Array(typeof count === "string" ? (count as any) * 1 : 2000)
            .fill(0)
            .map((item, index) => React.createElement(TextField))
    );
}

function render(): void {
    ReactDOM.render(React.createElement(renderItems), root);
}

render();
