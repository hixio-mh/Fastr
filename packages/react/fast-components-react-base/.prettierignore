coverage/*
dist/*
www/*
src/**/.tmp/