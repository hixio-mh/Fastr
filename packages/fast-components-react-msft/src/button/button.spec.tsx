import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, mount, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import { ButtonHTMLTags } from "@microsoft/fast-components-react-base";
import { IButtonClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
import {
    Button,
    ButtonAppearance,
    ButtonProps,
    IButtonHandledProps,
    IButtonManagedClasses,
    IButtonUnhandledProps,
    IMSFTButtonClassNameContract
} from "./button";

/*
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("button snapshots", (): void => {
    generateSnapshots(examples);
});

describe("button", (): void => {
    const Component: React.ComponentClass<IButtonHandledProps & IButtonUnhandledProps> = examples.component;

    const href: string = "https://www.microsoft.com";

    const beforeSlotExample: JSX.Element = (
        <div
            className={"slotBefore"}
            key={"Slot"}
            slot="before"
        >
            Before slot
        </div>
    );

    const afterSlotExample: JSX.Element = (
        <div
            className={"slotAfter"}
            key={"afterSlot"}
            slot="after"
        >
            After slot
        </div>
    );

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: IButtonHandledProps = {
            href
        };

        const unhandledProps: IButtonUnhandledProps = {
            "aria-hidden": true
        };

        const props: IButtonHandledProps & IButtonUnhandledProps = {...handledProps, ...unhandledProps};

        const rendered: any = shallow(
            <Component {...props}/>
        );

        const button: any = rendered.first().shallow();

        expect(button.prop("aria-hidden")).toEqual(true);
    });

    /* tslint:disable-next-line */
    test("should apply a 'primary' html class when appearance is primary", () => {
        const props: IButtonHandledProps = {
            appearance: ButtonAppearance.primary
        };

        const rendered: any = shallow(
            <Component {...props}/>
        );

        const button: any = rendered.first().shallow();
        // Get the expected className value from the list of generated managed classes
        const expectedClassName: string = button.instance().props.managedClasses.button__primary;

        expect(button.instance().props.appearance).toEqual(ButtonAppearance.primary);
        // Generated managedClass should be passed to className
        expect(button.prop("className")).toBe(expectedClassName);
    });

    /* tslint:disable-next-line */
    test("should apply an 'outline' html class when appearance is outline", () => {
        const props: IButtonHandledProps = {
            appearance: ButtonAppearance.outline
        };

        const rendered: any = shallow(
            <Component {...props}/>
        );

        const button: any = rendered.first().shallow();
        // Get the expected className value from the list of generated managed classes
        const expectedClassName: string = button.instance().props.managedClasses.button__outline;

        expect(button.instance().props.appearance).toEqual(ButtonAppearance.outline);
        // Generated managedClass should be passed to className
        expect(button.prop("className")).toBe(expectedClassName);
    });

    /* tslint:disable-next-line */
    test("should apply a 'lightweight' html class when appearance is lightweight", () => {
        const props: IButtonHandledProps = {
            appearance: ButtonAppearance.lightweight
        };

        const rendered: any = shallow(
            <Component {...props}/>
        );

        const button: any = rendered.first().shallow();
        // Get the expected className value from the list of generated managed classes
        const expectedClassName: string = button.instance().props.managedClasses.button__lightweight;

        expect(button.instance().props.appearance).toEqual(ButtonAppearance.lightweight);
        // Generated managedClass should be passed to className
        expect(button.prop("className")).toBe(expectedClassName);
    });

    /* tslint:disable-next-line */
    test("should apply a 'justified' html class when appearance is justified", () => {
        const props: IButtonHandledProps = {
            appearance: ButtonAppearance.justified
        };

        const rendered: any = shallow(
            <Component {...props}/>
        );

        const button: any = rendered.first().shallow();
        // Get the expected className value from the list of generated managed classes
        const expectedClassName: string = button.instance().props.managedClasses.button__justified;

        expect(button.instance().props.appearance).toEqual(ButtonAppearance.justified);
        // Generated managedClass should be passed to className
        expect(button.prop("className")).toBe(expectedClassName);
    });

    test("should set a custom class name when passed", () => {
        const customClassNameString: string = "customClassName";
        const rendered: any = shallow(
            <Component className={customClassNameString} />
        );

        const button: any = rendered.first().shallow();

        expect(button.prop("className")).toEqual(customClassNameString);
    });

    test("should set a custom class name and 'justified' class name when appearance is justified and a custom class is passed", () => {
        const customClassNameString: string = "customClassName";
        const rendered: any = shallow(
            <Component appearance={ButtonAppearance.justified} className={customClassNameString} />
        );

        const button: any = rendered.first().shallow();
        const expectedClassName: string = `${button.instance().props.managedClasses.button__justified} ${customClassNameString}`;

        expect(button.prop("className")).toEqual(expectedClassName);
    });

    /* tslint:disable-next-line */
    test("should add a child element with the slot prop set to 'before' into the before slot location", () => {
        const props: IButtonHandledProps = {
            appearance: ButtonAppearance.lightweight,
            href: "#",
            children: ["foo", beforeSlotExample]
        };

        const rendered: any = mount(
            <Component {...props} />
        );

        expect(rendered.instance().props.children[1].props.slot).toBe("before");
        expect(rendered.find("div.slotBefore").length).toBe(1);
    });

    /* tslint:disable-next-line */
    test("should add a child element with the slot prop set to 'after' into the after slot location", () => {
        const props: IButtonHandledProps = {
            appearance: ButtonAppearance.lightweight,
            href: "#",
            children: ["foo", afterSlotExample]
        };

        const rendered: any = mount(
            <Component {...props} />
        );

        expect(rendered.instance().props.children[1].props.slot).toBe("after");
        expect(rendered.find("div.slotAfter").length).toBe(1);
    });
});
