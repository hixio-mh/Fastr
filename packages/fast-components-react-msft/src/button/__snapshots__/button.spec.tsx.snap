// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshots Button: 0 1`] = `
<button
  className="button-0-1-8"
  data-sketch-symbol="Button - secondary"
  href={null}
>
  <span
    className="button_contentRegion-0-1-13"
  >
    Secondary button
  </span>
</button>
`;

exports[`button snapshots Button: 1 1`] = `
<button
  className="button-0-1-22 button__primary-0-1-23"
  data-sketch-symbol="Button - primary"
  href={null}
>
  <span
    className="button_contentRegion-0-1-27"
  >
    Primary button
  </span>
</button>
`;

exports[`button snapshots Button: 2 1`] = `
<button
  className="button-0-1-36 button__outline-0-1-38"
  data-sketch-symbol="Button - outline"
  href={null}
>
  <span
    className="button_contentRegion-0-1-41"
  >
    Outline button
  </span>
</button>
`;

exports[`button snapshots Button: 3 1`] = `
<button
  className="button-0-1-50 button__lightweight-0-1-53"
  data-sketch-symbol="Button - lightweight"
  href={null}
>
  <span
    className="button_contentRegion-0-1-55"
  >
    Lightweight button
  </span>
</button>
`;

exports[`button snapshots Button: 4 1`] = `
<button
  className="button-0-1-64 button__justified-0-1-68"
  href={null}
>
  <span
    className="button_contentRegion-0-1-69"
  >
    Justified button
  </span>
</button>
`;

exports[`button snapshots Button: 5 1`] = `
<a
  className="button-0-1-78"
  href="#"
>
  <span
    className="button_contentRegion-0-1-83"
  >
    Anchor
  </span>
</a>
`;

exports[`button snapshots Button: 6 1`] = `
<a
  className="button-0-1-92"
  href="#"
>
  <div
    slot="before"
  >
    &lt;
  </div>
  <span
    className="button_contentRegion-0-1-97"
  >
    Before slot only
  </span>
</a>
`;

exports[`button snapshots Button: 7 1`] = `
<a
  className="button-0-1-106"
  href="#"
>
  <span
    className="button_contentRegion-0-1-111"
  >
    After slot only
  </span>
  <div
    slot="after"
  >
    &gt;
  </div>
</a>
`;

exports[`button snapshots Button: 8 1`] = `
<a
  className="button-0-1-120"
  href="#"
>
  <div
    slot="before"
  >
    &lt;
  </div>
  <span
    className="button_contentRegion-0-1-125"
  >
    Both slots
  </span>
  <div
    slot="after"
  >
    &gt;
  </div>
</a>
`;

exports[`button snapshots Button: 9 1`] = `
<a
  className="button-0-1-134"
  href="#"
>
  <div
    slot="before"
  >
    &lt;
  </div>
  <div
    slot="before"
  >
    &lt;
  </div>
  <span
    className="button_contentRegion-0-1-139"
  >
    Mutliple both slots
  </span>
  <div
    slot="after"
  >
    &gt;
  </div>
  <div
    slot="after"
  >
    &gt;
  </div>
</a>
`;

exports[`button snapshots Button: 10 1`] = `
<a
  className="button-0-1-148"
  href="#"
>
  <div
    slot="before"
  >
    &lt;
  </div>
  <div
    slot="before"
  >
    &lt;
  </div>
  <span
    className="button_contentRegion-0-1-153"
  />
  <div
    slot="after"
  >
    &gt;
  </div>
  <div
    slot="after"
  >
    &gt;
  </div>
</a>
`;

exports[`button snapshots Button: 11 1`] = `
<button
  className="button-0-1-162 button__disabled-0-1-168"
  data-sketch-symbol="Button - secondary disabled"
  disabled={true}
  href={null}
>
  <span
    className="button_contentRegion-0-1-167"
  >
    Secondary button (disabled)
  </span>
</button>
`;

exports[`button snapshots Button: 12 1`] = `
<button
  className="button-0-1-176 button__disabled-0-1-182 button__primary-0-1-177"
  data-sketch-symbol="Button - primary disabled"
  disabled={true}
  href={null}
>
  <span
    className="button_contentRegion-0-1-181"
  >
    Primary button (disabled)
  </span>
</button>
`;

exports[`button snapshots Button: 13 1`] = `
<button
  className="button-0-1-190 button__disabled-0-1-196 button__outline-0-1-192"
  data-sketch-symbol="Button - outline disabled"
  disabled={true}
  href={null}
>
  <span
    className="button_contentRegion-0-1-195"
  >
    Outline button (disabled)
  </span>
</button>
`;

exports[`button snapshots Button: 14 1`] = `
<button
  className="button-0-1-204 button__disabled-0-1-210 button__lightweight-0-1-207"
  data-sketch-symbol="Button - lightweight disabled"
  disabled={true}
  href={null}
>
  <span
    className="button_contentRegion-0-1-209"
  >
    Lightweight button (disabled)
  </span>
</button>
`;

exports[`button snapshots Button: 15 1`] = `
<button
  className="button-0-1-218 button__disabled-0-1-224 button__justified-0-1-222"
  disabled={true}
  href={null}
>
  <span
    className="button_contentRegion-0-1-223"
  >
    Justified button (disabled)
  </span>
</button>
`;

exports[`button snapshots Button: 16 1`] = `
<a
  aria-disabled={true}
  className="button-0-1-232 button__disabled-0-1-238"
  href="#"
>
  <span
    className="button_contentRegion-0-1-237"
  >
    Anchor (disabled)
  </span>
</a>
`;
